vm/devices/storage/disk_nvme/nvme_driver/src/driver.rs

```diff
@@ -664,6 +667,20 @@ impl<T: DeviceBacking> NvmeDriver<T> {
             .attach_pending_buffers()
             .context("failed to restore allocations")?;
 
+        // Log admin queue restore details
+        if let Some(ref admin_state) = saved_state.worker_data.admin {
+            tracing::info!(
+                "restoring admin queue from: {{sqid: {}, cqid: {}, pending_commands_count: {}, sq_head: {}, cq_head: {}}}",
+                admin_state.handler_data.sq_state.sqid,
+                admin_state.handler_data.cq_state.cqid,
+                admin_state.handler_data.pending_cmds.commands.len(),
+                admin_state.handler_data.sq_state.head,
+                admin_state.handler_data.cq_state.head,
+            );
```

Comment:
Prefer to use fields rather than formatting. This results in semi-structured data that is easier for us to consume at scale. You may need the `?` (debug) or `%` (display) sigil, but likely not for these simple types. If you need to use it, it would be like: `sqid = ?admin_state...`.

```suggestion
            tracing::info!(
                sqid = admin_state.handler_data.sq_state.sqid,
                cqid = admin_state.handler_data.cq_state.cqid,
                pending_commands_count = admin_state.handler_data.pending_cmds.commands.len(),
                sq_head = admin_state.handler_data.sq_state.head,
                cq_head = admin_state.handler_data.cq_state.head,
                "restoring admin queue",
            );
```